WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+                }
+            }
+            .navigationTitle(Localization.addPackage)
+            .navigationBarTitleDisplayMode(.inline)
+        }
+        .navigationViewStyle(.stack)
+    }
+
+    @ViewBuilder
+    private var selectedPackageTypeView: some View {
+        ScrollView {
+            switch selectedPackageType {
+            case .custom:
+                customPackageView
+            case .carrier:
+                carrierPackageView
+            case .saved:
+                savedPackageView
+            }
+        }
+    }
+
+    enum PackageType: CaseIterable {
+        case box, envelope
+        var name: String {
+            switch self {
+            case .box:
+                return Localization.box
+            case .envelope:
+                return Localization.envelope
+            }
+        }
+    }
+
+    @State var packageType: PackageType = .box
+    @State var showSaveTemplate: Bool = false
+    @State var length: String = ""
+    @State var fieldValues: [WooShippingAddPackageDimensionView.DimensionType: String] = [:]
+
+    private var customPackageView: some View {
+        VStack(alignment: .leading) {
+            HStack {
+                Text(Localization.packageType)
+                    .font(.subheadline)
+                Spacer()
+            }
+            // type selection
+            Menu {
+                // show selection
+                ForEach(PackageType.allCases, id: \.self) { option in
+                    Button {
+                        packageType = option
+                    } label: {
+                        Text(option.name)
+                            .font(.body)
+                    }
+                }
+            } label: {
+                HStack {
+                    // text
+                    Text(packageType.name)
+                    // arrows
+                    Spacer()
+                    Image(systemName: "chevron.up.chevron.down")
+                }
+                .padding()
+            }
+            .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
+            HStack(spacing: 8) {
+                ForEach(WooShippingAddPackageDimensionView.DimensionType.allCases, id: \.self) { dimensionType in
+                    WooShippingAddPackageDimensionView(dimensionType: dimensionType, fieldValue: Binding(get: {
+                        return self.fieldValues[dimensionType] ?? ""
+                    }, set: { value in
+                        self.fieldValues[dimensionType] = value
+                    }))
+                }
+            }
+            Toggle(isOn: $showSaveTemplate) {
+                Text(Localization.saveNewPackageTemplate)
+                    .font(.subheadline)
+            }
+            if showSaveTemplate {
+                Button(Localization.savePackageTemplate) {
+                    // save template
+                }
+                .buttonStyle(SecondaryButtonStyle())
+            }
```

Comment:
Based on the designs, we also need a text field here to enter the package name. It looks like when toggled on, this text field should get focus.

Reply:
started with this in https://github.com/woocommerce/woocommerce-ios/pull/14086